### ISSUE TITLE
XPath Assignment Rewriting

### ISSUE THREAD

**Issue by ebeshero** [ebeshero/newtFire-webDev#5]

**Comment by ebeshero** [comment]
## File chosen

Text for base file (to replace badhamlet): Actually...[my own Syllabus XML file](http://newtfire.org/dh/dhCDA-2015.xml): because:
 a) it's not the same structure as the play at all, but has a similar complex hierarchy,
b) it has date strings in its attributes for each day on the schedule in the XML table, and we can tokenize on the portions of the dates to pull up, say, all the September assignments. We can also run XPaths that tokenize on date _and_ search for some specific string in the content or in a neighboring element's attribute, such as `@type="assign.`
c) We can do something similar to question 1 on [Obdurodon's assignment](http://dh.obdurodon.org/xpath-assignment04.html), but ask students to generate a semicolon-separated list of all the highlighted assignment names...and then move on to concatenate the output with dates. This could be made more challenging by asking for the date on which they're _due_--which would demand looking for the date on the next row of the XML table!
d) For the challenge questions, try asking people to find all the `<ref @target="url"/>` elements where the url contains/matches "pitt.edu" somewhere inside, and output a neatly numbered list of distinct addresses.

**Comment by ebeshero** [comment]
@alexkrongel Okay, here is a challenge task! I was going to do this myself, but I think maybe, in the interests of time, you could start it while I keep hunting for suitable files for the XSLT assignments. Alex, would you be willing to try a creative rewrite of [Obdurodon's XPath #4](http://dh.obdurodon.org/xpath-assignment04.html) based on the text and plan I posted here? (See links above in the first post of this thread). To start, I'll save Obdurodon's file here and write in some comments, and ping you from the file. Whatever you can do here to get this started today will help to save us some time!

**Comment by ebeshero** [comment]
@alexkrongel I've pinged you from the most recent commit of XPath Exercise 4--and I'll move my requests here so it's a little more permanent. 
Here's the commit: https://github.com/ebeshero/newtFire-webDev/commit/617858dddcb1c4993608428071da8aea9fd9e8d4

@RJP43 and @andrewntz23 Just to be clear, here's how we can work on this and other HTML assignment files:
1) Clone this repository if you haven't already, and find your way to the relevant file locally. (The file directory is huge and I apologize, but the file names are predictable at least and I'll deliver you the file names.) (For example, all the XPath assignments are named consistently, and this one we're working on is named XPathExercise4.html .) 

2) Go ahead and edit away, changing the tasks so they work with the new file we've selected, and changing instructions as needed. 

3) Push your changes through to GitHub as you work. 

Let me know if you have any questions or problems! And we can ping each other from commits like this (and make line comments if you like) or work over in Issues for this assignment. 
Thanks, everyone!

**Comment by alexkrongel** [comment]
@ebeshero I've created some questions that I think are different enough but still cover the same concepts and pushed them. Please let me know if they need to be made more or less difficult. I'm still thinking about possible challenge questions!

**Comment by ebeshero** [comment]
@alexkrongel Yay! Thank you--I've not had a chance to look at them yet, but you've pushed them here, right? Meanwhile, here's a possible thing we can ask: Have the students grab the dates on the syllabus and output all the Mondays, NOT by looking that up on the syllabus but by using format-date(). We need to use XPath 3.0 to do this--but let's give it a try and see if it's possible!

**Comment by ebeshero** [comment]
@alexkrongel I've just read your questions. Thank you-they're very good and manageable! We need some challenge questions, and the Monday-dates could be one of them...

**Comment by alexkrongel** [comment]
@ebeshero Ok great! I'll add a Monday-dates question and comment in the answers so others can check it this afternoon or tonight.

**Comment by alexkrongel** [comment]
@ebeshero So I've been playing around with this and I may be overlooking something, but I'm not sure format-date() can handle dates as they are on the syllabus right now. The expression //div[@type='day']//date[format-date(., F)] returns the error "Invalid Date M 08-31 : Non-numeric year component" - I think the calendar it's pulling from (F signifies output for day of the week) requires the year. I'll keep trying to see if I can first reformat the date to 2015-[M01]-[D01] and then ask for day of the week, but I wanted to alert you to the difficulty now.

**Comment by ebeshero** [comment]
Thanks @alexkrongel : How did the date-matching work out? I might have had better luck with grabbing ISO dates in JavaScript.

**Comment by ebeshero** [comment]
@alexkrongel Aha! I see how it worked out--I'm looking at your challenge question now. This looks great. Thank you!

**Comment by ebeshero** [comment]
@alexkrongel I've posted our new [XPath Exercise 4](http://newtfire.org/dh/XpathExercise4.html) on newtFire, after some modifications! I couldn't work out the last challenge question with format-date() myself, and found very little info on it as an XPath function (and much more on similar things in JavaScript and PHP), though I'm tired and might have missed something, so I decided to leave it out. If you found a solution, want to share it?
